src/AgentModel.cpp

```diff
@@ -655,35 +774,64 @@ void AgentModel::consciousFollow() {
     // get pointer to targets
     auto t = _input.targets;
 
-    // calculate net distance for following
-    unsigned long im = agent_model::NOT;
+    // calculate net distance for following targets
+    unsigned long im = agent_model::NOT;        // target on ego lane
+    unsigned long im_n_l = agent_model::NOT;    // target on neigbouring lane
+
+    int insert_idx;
     for (unsigned long i = 0; i < agent_model::NOT; ++i) {
 
         // ignore non-relevant targets (, ds < 0, other lane)
-        if (t[i].id == 0 || std::isinf(t[i].ds) || t[i].ds < 0.0 || t[i].lane != 0)
-            continue;
-
-        // check if distance is smaller
-        if (im == agent_model::NOT || t[im].ds > t[i].ds)
-            im = i;
+        if (t[i].id == 0 || std::isinf(t[i].ds) || t[i].ds < 0.0) continue;
+        
+        
+        if (t[i].lane == 0) {
 
+            // check if distance is smaller
+            if (im == agent_model::NOT || t[im].ds > t[i].ds)
+                im = i;
+        }
+        
+        // compute target of neigbouring lane only during lane changes
+        if (_lane_change_process_interval.getFactor() > 0 && t[i].lane == _state.decisions.laneChange) 
+        {
+            // check if distance is smaller
+            if (im_n_l == agent_model::NOT || t[im_n_l].ds > t[i].ds)
+                im_n_l = i;
+        }
     }
+    
+    // instantiate distance and velocity
+    double ds = INFINITY, v = 0.0, factor = 1;
```

Comment:
Just a minor remark: Factor is immediatly overwritten 3 lines after that

Reply:
Fixed